Wikipedia/Code/TalkPageViewController+FindInPage.swift

```diff
@@ -0,0 +1,104 @@
+import Foundation
+
+extension TalkPageViewController {
+
+    // MARK: - Overrides
+
+    override var canBecomeFirstResponder: Bool {
+        return findInPageState.keyboardBar != nil
+    }
+
+    override var inputAccessoryView: UIView? {
+        return findInPageState.keyboardBar
+    }
+
+    // MARK: - Presentation
+
+    fileprivate func createFindInPageViewIfNecessary() {
+        guard findInPageState.keyboardBar == nil else {
+            return
+        }
+
+        let keyboardBar = FindAndReplaceKeyboardBar.wmf_viewFromClassNib()!
+        keyboardBar.delegate = self
+        keyboardBar.apply(theme: theme)
+        findInPageState.keyboardBar = keyboardBar
+    }
+
+    func showFindInPage() {
+        createFindInPageViewIfNecessary()
+        becomeFirstResponder()
+        findInPageState.keyboardBar?.show()
+    }
+
+    func hideFindInPage() {
+        findInPageState.reset(viewModel.topics)
+        findInPageState.keyboardBar?.hide()
+        talkPageView.collectionView.reloadData()
+        resignFirstResponder()
+    }
+
+    // MARK: - Scroll to Element
+
+    func scrollToFindInPageResult(_ result: TalkPageFindInPageSearchController.SearchResult?) {
+        guard let result = result else { return }
+
+        talkPageView.collectionView.reloadData()
+
+        // TODO: - Fine tune position when scrolling
+        switch result.location {
+        case .topicTitle(topicIndex: let index, topicIdentifier: _):
+            let indexPath = IndexPath(row: index, section: 0)
+            talkPageView.collectionView.scrollToItem(at: indexPath, at: .top, animated: true)
+        case .topicLeadComment(topicIndex: let index, replyIdentifier: _):
+            let commentViewModel = viewModel.topics[index].leadComment
+            scrollToComment(commentViewModel: commentViewModel, animated: true)
```

Comment:
Just a heads up, unfortunately my `scrollToComment` method I made only considers comment views beneath the lead comment. So you might need to figure out an alternative way to scroll to the lead comment, or modify `scrollToComment` to consider the lead comment text view.

Reply:
Thanks for the heads up. Personally I view my logic here as in progress. Added that TODO previously as I intended to come back to refining this scrolling in a future PR.